tests/Sdk/Unit/Infrastructure/Service/TaskSet/TaskFromTaskSetBuilderTest.php

```diff
@@ -0,0 +1,201 @@
+            $this->createTaskSetOverrideMapFactoryMock(),
+        );
+
+        // Act
+        $this->expectException(InvalidArgumentException::class);
+        $taskSetBuilder->buildTaskFromTaskSet($taskSet, $existingTasks);
+    }
+
+    /**
+     * @return void
+     */
+    public function testBuildsTaskWhenSubTaskIsObject(): void
+    {
+        // Arrange
+        $command = $this->createCommandMock();
+        $placeholder = $this->createPlaceholderMock();
+        $task = $this->createTaskMock('taskId', $command, $placeholder);
+        $existingTasks = ['taskId' => $task];
+        $taskSet = $this->createTaskSetMock('taskSetId', null, $task);
+        $taskSetBuilder = new TaskFromTaskSetBuilder(
+            $this->createTaskSetPlaceholdersBuilderMock($placeholder),
+            $this->createTaskSetCommandsBuilderMock($command),
+            $this->createTaskSetOverrideMapFactoryMock(),
+        );
+
+        // Act
+        $task = $taskSetBuilder->buildTaskFromTaskSet($taskSet, $existingTasks);
+
+        // Assert
+        $this->assertSame($command, $task->getCommands()[0]);
+        $this->assertSame($placeholder, $task->getPlaceholders()[0]);
+        $this->assertSame('taskSetId', $task->getId());
+    }
+
+    /**
+     * @param string $taskSetId
+     * @param string|null $taskId
+     * @param \SprykerSdk\SdkContracts\Entity\TaskInterface|null $task
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\TaskSetInterface
+     */
+    protected function createTaskSetMock(string $taskSetId, ?string $taskId, ?TaskInterface $task): TaskSetInterface
+    {
+        $taskMock = $this->createMock(TaskSetInterface::class);
+
+        if ($taskId !== null) {
+            $taskMock->method('getSubTasks')->willReturn([$taskId]);
+        } else {
+            $taskMock->method('getSubTasks')->willReturn([$task]);
+        }
+
+        $taskMock->method('getId')->willReturn($taskSetId);
+        $taskMock->method('getShortDescription')->willReturn('');
+        $taskMock->method('getLifecycle')->willReturn($this->createMock(LifecycleInterface::class));
+        $taskMock->method('getVersion')->willReturn('');
+        $taskMock->method('getHelp')->willReturn(null);
+        $taskMock->method('getSuccessor')->willReturn(null);
+        $taskMock->method('isDeprecated')->willReturn(false);
+        $taskMock->method('isOptional')->willReturn(true);
+        $taskMock->method('getStages')->willReturn([]);
+
+        return $taskMock;
+    }
+
+    /**
+     * @param string $taskId
+     * @param \SprykerSdk\Sdk\Presentation\Ide\PhpStorm\Dto\CommandInterface $command
+     * @param \SprykerSdk\SdkContracts\Entity\PlaceholderInterface $placeholder
+     *
+     * @return \SprykerSdk\SdkContracts\Entity\TaskInterface
+     */
+    protected function createTaskMock(string $taskId, CommandInterface $command, PlaceholderInterface $placeholder): TaskInterface
+    {
+        $taskMock = $this->createMock(TaskSetInterface::class);
```

Comment:
The same as here https://github.com/spryker-sdk/sdk/pull/207/files#r970695350

Reply:
same